packages/smooth_app/lib/helpers/product_cards_helper.dart

```diff
@@ -113,35 +113,80 @@ List<ProductImageData> getProductMainImagesData(
   final bool includeOther = true,
 }) =>
     <ProductImageData>[
-      ProductImageData(
-        imageField: ImageField.FRONT,
-        imageUrl: product.imageFrontUrl,
-        title: appLocalizations.product,
-        buttonText: appLocalizations.front_photo,
-      ),
-      ProductImageData(
-        imageField: ImageField.INGREDIENTS,
-        imageUrl: product.imageIngredientsUrl,
-        title: appLocalizations.ingredients,
-        buttonText: appLocalizations.ingredients_photo,
-      ),
-      ProductImageData(
-        imageField: ImageField.NUTRITION,
-        imageUrl: product.imageNutritionUrl,
-        title: appLocalizations.nutrition,
-        buttonText: appLocalizations.nutrition_facts_photo,
-      ),
-      ProductImageData(
-        imageField: ImageField.PACKAGING,
-        imageUrl: product.imagePackagingUrl,
-        title: appLocalizations.packaging_information,
-        buttonText: appLocalizations.packaging_information_photo,
-      ),
+      getProductImageData(product, appLocalizations, ImageField.FRONT),
+      getProductImageData(product, appLocalizations, ImageField.INGREDIENTS),
+      getProductImageData(product, appLocalizations, ImageField.NUTRITION),
+      getProductImageData(product, appLocalizations, ImageField.PACKAGING),
       if (includeOther)
-        ProductImageData(
-          imageField: ImageField.OTHER,
-          imageUrl: null,
-          title: appLocalizations.more_photos,
-          buttonText: appLocalizations.more_photos,
-        ),
+        getProductImageData(product, appLocalizations, ImageField.OTHER),
     ];
+
+ProductImageData getProductImageData(
+  final Product product,
+  final AppLocalizations? appLocalizations,
```

Comment:
Why is this nullable

Reply:
The thing is that `ProductImageData` does 2 different things: it computes the image provider or url, and it populates the title and button labels.
In some cases we don't use the labels, therefore we don't need the `AppLocalizations`. I just wanted to avoid calling a `AppLocalizations.of(context)` just for something I don't care about - or maybe the `context` was not available there, don't remember.
Could be refactored later, especially now that I created specific methods for each use-case (button label, title label, url, provider).